python-backend/app/api/mines/mine/resources/mine_tenure_type_code.py

```diff
@@ -1,11 +1,16 @@
 from flask_restplus import Resource
 from app.extensions import api
+from app.api.mines.mine_api_models import MINE_TENURE_TYPE_CODE_MODEL
 from ....utils.access_decorators import requires_role_mine_view
 from ....utils.resources_mixins import UserMixin, ErrorMixin
 from ..models.mine_tenure_type_code import MineTenureTypeCode
 
+
 class MineTenureTypeCodeResource(Resource, UserMixin, ErrorMixin):
-    @api.doc(params={})
+    @api.doc(description='Returns all the mineral tenure type codes.')
     @requires_role_mine_view
+    @api.marshal_with(MINE_TENURE_TYPE_CODE_MODEL, code=200)
     def get(self):
-        return { 'options': MineTenureTypeCode.all_options() }
+        mineral_tenure_type_codes = MineTenureTypeCode.find_all()
+
+        return mineral_tenure_type_codes
```

Comment:
We've agreed to put a `records` envelope around our list responses, correct? Please add that here.

Also, I'm not sure that `.find_all()` is actually buying us anything over just `.query.all()`, so I'm curious if you have a reason for adding that method.

Finally, if you've removed the use of `.all_options()`, could you please remove that method from the model? thanks!

Reply:
I have removed the all_options function and renamed the find_all to find_all_active. As for why I did it as a model class method as far as i know we don't really have a standard on where we put the actual DB query code. Half is in the models and then some of it lives in the resources.